### PR TITLE
Made UI of search jobs page

### PR DESCRIPTION
**Pull request by lerabo**
localhost_3001_post-job (1)

### CODE REVIEW

**Comment by lyudad** [comment]
filters to wide, they take almost the same space that the job post list

**Comment by lyudad** [comment]
Снимок экрана 2022-09-08 в 18 08 48
inputs visually different, as a programmer, I understand why it like that, as user, I expect that the same design rules will be followed everywhere

**Comment by lerabo** [comment]
localhost_3001_post-job (3) Changed

File: src/components/WelcomePage/WelcomePageLayout.tsx [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
 	const { t } = useTranslation();
+	const navigate = useNavigate();
 
 	const Role = {
 		Freelancer: 'freelancer',
 		Client: 'client',
 	};
 
+	useEffect(() => {
+		if (user.role === null) {
```
Comment (lyudad) [inline src/components/WelcomePage/WelcomePageLayout.tsx]: !user.role - can we go this way?

File: src/components/freelancerJobs/freelancerPage.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+				/>
+				{e.name}
+			</Label>
+		));
+	}, [skillsOptions]);
+
+	const intitialState = {
+		value: '',
+		label: '',
+	};
```
Comment (lyudad) [inline src/components/freelancerJobs/freelancerPage.tsx]: first use state and use effect, then handlers functions and after render function, please put your mess in order
and constant should be initialising out of components as in other case initialising will happened with each rerender
also types are missed

File: src/components/freelancerJobs/freelancerPage.tsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+	const [slider, setSlider] = useState([0, 10000]);
+	const [search, setSearch]: [string, (search: string) => void] = useState('');
+	const [userChoice, setUserChoice] = useState(intitialState);
+	const [radio, setRadio] = useState('');
```
Comment (lyudad) [inline src/components/freelancerJobs/freelancerPage.tsx]: types for useState are missed

File: src/components/WelcomePage/WelcomePageLayout.tsx (hunk 1 above)
Comment (lerabo) [reply src/components/WelcomePage/WelcomePageLayout.tsx]: replaced

File: src/components/freelancerJobs/freelancerPage.tsx (hunk 3 above)
Comment (lerabo) [reply src/components/freelancerJobs/freelancerPage.tsx]: added

File: src/components/freelancerJobs/freelancerPage.tsx (hunk 2 above)
Comment (lerabo) [reply src/components/freelancerJobs/freelancerPage.tsx]: fixed

File: src/components/WelcomePage/WelcomePageLayout.tsx [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
+	const navigate = useNavigate();
 
 	const Role = {
 		Freelancer: 'freelancer',
 		Client: 'client',
 	};
 
+	useEffect(() => {
+		if (!user.role) {
+			navigate('/role-selection');
```
Comment (lyudad) [inline src/components/WelcomePage/WelcomePageLayout.tsx]: all path move to const file routes.js or something like that

Comment (lyudad) [inline src/components/WelcomePage/WelcomePageLayout.tsx]: move all routes to /constant/routes.ts

File: src/components/freelancerJobs/freelancerPage.tsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+					checked={e.value}
+					autoComplete="off"
+					onChange={() => onSkillsChange(i)}
+				/>
+				{e.name}
+			</Label>
+		));
+	}, [skillsOptions]);
+
+	const [search, setSearch] = useState<string>('');
```
Comment (lyudad) [inline src/components/freelancerJobs/freelancerPage.tsx]: it was not fixed. I explaine the order
hooks, handles, renders
now its mess

File: src/components/freelancerJobs/freelancerPage.tsx [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+										</Link>
+									</Li>
+								))}
+						</ul>
+					</ColumnBig>
+				</>
+			)}
+		</>
+	);
+};
```
Comment (lyudad) [inline src/components/freelancerJobs/freelancerPage.tsx:145]: component is too big, move logic to custom hook or/and device to smaller components

File: src/components/WelcomePage/WelcomePageLayout.tsx (hunk 4 above)
Comment (lerabo) [reply src/components/WelcomePage/WelcomePageLayout.tsx]: moved

Comment (lerabo) [reply src/components/WelcomePage/WelcomePageLayout.tsx]: moved

File: src/components/freelancerJobs/freelancerPage.tsx (hunk 5 above)
Comment (lerabo) [reply src/components/freelancerJobs/freelancerPage.tsx]: changed

File: src/components/freelancerJobs/freelancerPage.tsx (hunk 6 above)
Comment (lerabo) [reply src/components/freelancerJobs/freelancerPage.tsx:145]: added smaller components

**Pull request merged**